src/device/resource_allocator_v4.c

```diff
@@ -198,15 +212,21 @@ static void compute_domain_resources(const struct device *domain)
 {
 	const struct device *child;
 	const int print_depth = 1;
+	struct bus *bus = domain->link_list;
 
 	if (domain->link_list == NULL)
 		return;
 
-	for (child = domain->link_list->children; child; child = child->sibling) {
+	while(bus && bus->children == NULL)
```

Comment:
What will happen if bus is NULL instead of bus->children? `bus = bus->next;` will just access NULL pointer

Reply:
That is why there is test for `bus` before testing `bus->children`, neither the second test nor `bus->next` won't be accessed when `bus` is NULL. However, it does lack test for `bus` being NULL after this loop, where it blindly accesses `bus->children` in next `for` loop in most cases.